dnamic_analysis/output.py

```diff
@@ -0,0 +1,323 @@
+                                                                    avg_sum,avg_len,avg_overall,avg_overall/365,percent_len,all_len,percent_overall)
+        self._excel_object.write(self._worksheet, self._col, 0, data, 'row1')
+        self._excel_object.write(self._worksheet, self._col, 1, 'Expired Domain Privileged IDs', 'header')
+        row = 2
+        for username,_,_ in max_sorted:
+            self._excel_object.write(self._worksheet, self._col, row, username)
+            row += 1
+        self._col += 1
+        self._excel_object.save(self._workbook)
+
+
+    #########################################
+    ## Unique Expired Local Privileged IDs ##
+    #########################################
+    def local_expired(self,max_sorted,avg_sum,avg_len,avg_overall,percent_len,all_len,percent_overall,sqlcount,unique_count):
+        # Write bulk data to Excel workbook
+        data = 'Oldest Non-Compliant Username: {}\n' \
+                'Max Password Age: {} days ({:.1f} years)\n' \
+                'Total Avg Password Age: {:.2f} / {} = {:.2f} days ({:.1f} years)\n' \
+                'Total Percent Non-Compliant: {} / {} = {:.2%}\n' \
+                'Total Unique Local Privileged IDs: {}\n' \
+                'Total Unique Local Privileged ID Names: {}'.format(max_sorted[0][0],max_sorted[0][3],max_sorted[0][3]/365,
+                                                                avg_sum,avg_len,avg_overall,avg_overall/365,percent_len,
+                                                                all_len,percent_overall,sqlcount,unique_count)
+        self._excel_object.write(self._worksheet, self._col, 0, data, 'row1')
+        self._excel_object.write(self._worksheet, self._col, 1, 'Unique Expired Local Privileged IDs', 'header')
+        row = 2
+        used_usernames = []
+        for username,_,_,_ in max_sorted:
+            if username not in used_usernames:
+                self._excel_object.write(self._worksheet, self._col, row, username)
+                used_usernames.append(username)
+                row += 1
+        self._col += 1
+        self._excel_object.save(self._workbook)
+
+
+    #####################################################
+    ## Expired Local Admins Total w/ Machine Addresses ##
+    #####################################################
+    def local_expired_machines(self,max_grouped,count_accounts,percent_accounts):
+        self._excel_object.write(self._worksheet, self._col, 1, 'Expired Local Admins Total w/ Machine Addresses', 'header')
+        count = 0
+        row = 2
+        for value in max_grouped.items():
+            for key in value:
+                if not isinstance(key, str):
+                    if isinstance(key, list):
+                        for machine in key:
+                            self._excel_object.write(self._worksheet, self._col, row, machine)
+                            count = len(key)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable 'count' is assigned to but never used](https://app.codacy.com/app/infamousjoeg/DNAmicAnalysis/pullRequest?prid=4572226)